services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java

```diff
@@ -1145,13 +1146,36 @@ Optional<String> getTopicForCurrentPushJob(
       latestTopic = Optional.of(versionTopics.get(0));
     }
 
+    Map<String, String> futureVersions = getFutureVersionsForMultiColos(clusterName, storeName);
+    for (Map.Entry<String, String> entry: futureVersions.entrySet()) {
+      ControllerClient controllerClient = getVeniceHelixAdmin().getControllerClientMap(clusterName).get(entry.getKey());
+      StoreResponse storeResponse = controllerClient.retryableRequest(5, c -> c.getStore(storeName));
+      StoreInfo store = storeResponse.getStore();
+      int futureVersion = Integer.parseInt(entry.getValue());
```

Comment:
Can `Integer.parseInt` throw exception for invalid `getValue()`?
Or is it guaranteed to be a valid input for the function call.

Reply:
Its set to -1 for error case, so we will always parse the string to int.